### ISSUE TITLE
Exception: PyMNNInterpreter_new: NetInstance::createFromFile failed

### ISSUE THREAD

**Issue by yjiangling** [alibaba/MNN#1477]
# 平台(如果交叉编译请再附上交叉编译目标平台):  
# Platform(Include target platform as well if cross-compiling):  

# Github版本:
# Github Version:

直接下载ZIP包请提供下载日期以及压缩包注释里的git版本(可通过``7z l zip包路径``命令并在输出信息中搜索``Comment`` 获得，形如``Comment = bc80b11110cd440aacdabbf59658d630527a7f2b``)。 git clone请提供 ``git commit`` 第一行的commit id

Provide date (or better yet, git revision from the comment section of the zip. Obtainable using ``7z l PATH/TO/ZIP`` and search for ``Comment`` in the output) if downloading source as zip,otherwise provide the first commit id from the output of ``git commit``

# 编译方式:
# Compiling Method

```
请在这里粘贴cmake参数或使用的cmake脚本路径以及完整输出
Paste cmake arguments or path of the build script used here as well as the full log of the cmake proess here or pastebin
```

# 编译日志:
# Build Log:

```
粘贴在这里
Paste log here or pastebin
```
![image](https://user-images.githubusercontent.com/42861880/118245365-498cde80-b4d3-11eb-9d80-6244aca8d73b.png)

Anyone can give me some help? Thanks  a lot.

**Comment by jxt1234** [comment]
For control flow model, use MNN.nn.load_module_from_file to load it.

**Comment by yjiangling** [comment]
> For control flow model, use MNN.nn.load_module_from_file to load it.

感谢解答，大神，又出现这个错误是怎么回事？麻烦帮我看看，谢谢~

![image](https://user-images.githubusercontent.com/42861880/118583858-50b03700-b7c8-11eb-9f6e-936b7a42ba90.png)


这是我推理部分的代码：

def inference(files):
    """ inference rnnt using a specific wave """
    interpreter = MNN.nn.load_module_from_file('rnnt.mnn', ['xs', 'xlen'], ['logits'])
    interpreter.cache()
    session = interpreter.createSession()
    input_tensor = interpreter.getSessionInputAll(session)
    xs,xlen=get_feature(files)
    interpreter.resizeTensor(input_tensor['xs'], {1, xs.shape[1], 400})
    interpreter.resizeTensor(input_tensor['xlen'], {xlen.shape[0]})
    interpreter.resizeSession(session)
    input_xs = MNN.Tensor(xs.shape, MNN.Halide_Type_Float,\
                    xs, MNN.Tensor_DimensionType_Tensorflow)
    input_xlen=MNN.Tensor(xlen.shape, MNN.Halide_Type_Int,\
                    xlen,MNN.Tensor_DimensionType_Tensorflow)
    input_tensor['xs'].copyFrom(input_xs)
    input_tensor['xlen'].copyFrom(input_xlen)
    interpreter.runSession(session)
    output_tensor = interpreter.getSessionOutput(session)
    tmp_output = MNN.Tensor((xs.shape[0],50), MNN.Halide_Type_Int, \
        np.ones([xs.shape[0], 50]).astype(np.int32), MNN.Tensor_DimensionType_Tensorflow)
    output_tensor.copyToHostTensor(tmp_output) 
    print("output result: ",tmp_output.getData())
    interpreter.removeCache()

if __name__ == "__main__":
    inference(['1.wav'])

**Comment by yjiangling** [comment]
> For control flow model, use MNN.nn.load_module_from_file to load it.

顺便请问一下，为什么一直报这个 Error for binary op: input0's type != input1's type？有影响吗？模型转换和推理时都会出现，但是模型转换最后还是显示done.

**Comment by yjiangling** [comment]
> For control flow model, use MNN.nn.load_module_from_file to load it.

大神，为什么同一个模型，用前一个版本的MNN转换就没问题，用当前最新版本转换就会报这个错误

![image](https://user-images.githubusercontent.com/42861880/118625758-dc43bb00-b7fc-11eb-9802-c4d15ecadae7.png)

**Comment by jxt1234** [comment]
这个错误是因为在没有输入 shape 的情况，mnn 作图优化需要计算 tensor 的 shape ，计算失败则一些图优化不再执行，不影响最终转换结果。另外你的 python 脚本写的有问题，参考这个

import numpy as np
import MNN
import cv2
import sys
def inference():
    """ inference mobilenet_v1 using a specific picture """
    net = MNN.nn.load_module_from_file(sys.argv[1], ["data"], ["prob"])
    image = cv2.imread(sys.argv[2])
    #cv2 read as bgr format
    image = image[..., ::-1]
    #change to rgb format
    image = cv2.resize(image, (224, 224))
    #resize to mobile_net tensor size
    image = image - (103.94, 116.78, 123.68)
    image = image * (0.017, 0.017, 0.017)
    #change numpy data type as np.float32 to match tensor's format
    image = image.astype(np.float32)
    #Make var to save numpy
    input_var = MNN.expr.placeholder([1, 224, 224, 3], MNN.expr.NHWC)
    input_var.write(image)
    #cv2 read shape is NHWC, Module's need is NC4HW4, convert it
    input_var = MNN.expr.convert(input_var, MNN.expr.NC4HW4)
    #inference
    output_var = net.forward(input_var)
    #the output from net may be NC4HW4, turn to linear layout
    output_var = MNN.expr.convert(output_var, MNN.expr.NHWC)
    print("output belong to class: {}".format(np.argmax(output_var.read())))

**Comment by yjiangling** [comment]
> 这个错误是因为在没有输入 shape 的情况，mnn 作图优化需要计算 tensor 的 shape ，计算失败则一些图优化不再执行，不影响最终转换结果。另外你的 python 脚本写的有问题，参考这个
> 
> import numpy as np
> import MNN
> import cv2
> import sys
> def inference():
> """ inference mobilenet_v1 using a specific picture """
> net = MNN.nn.load_module_from_file(sys.argv[1], ["data"], ["prob"])
> image = cv2.imread(sys.argv[2])
> #cv2 read as bgr format
> image = image[..., ::-1]
> #change to rgb format
> image = cv2.resize(image, (224, 224))
> #resize to mobile_net tensor size
> image = image - (103.94, 116.78, 123.68)
> image = image * (0.017, 0.017, 0.017)
> #change numpy data type as np.float32 to match tensor's format
> image = image.astype(np.float32)
> #Make var to save numpy
> input_var = MNN.expr.placeholder([1, 224, 224, 3], MNN.expr.NHWC)
> input_var.write(image)
> #cv2 read shape is NHWC, Module's need is NC4HW4, convert it
> input_var = MNN.expr.convert(input_var, MNN.expr.NC4HW4)
> #inference
> output_var = net.forward(input_var)
> #the output from net may be NC4HW4, turn to linear layout
> output_var = MNN.expr.convert(output_var, MNN.expr.NHWC)
> print("output belong to class: {}".format(np.argmax(output_var.read())))

好的，非常感谢~

**Comment by Hanson0910** [comment]
> 这个错误是因为在没有输入 shape 的情况，mnn 作图优化需要计算 tensor 的 shape ，计算失败则一些图优化不再执行，不影响最终转换结果。另外你的 python 脚本写的有问题，参考这个
> 
> import numpy as np import MNN import cv2 import sys def inference(): """ inference mobilenet_v1 using a specific picture """ net = MNN.nn.load_module_from_file(sys.argv[1], ["data"], ["prob"]) image = cv2.imread(sys.argv[2]) #cv2 read as bgr format image = image[..., ::-1] #change to rgb format image = cv2.resize(image, (224, 224)) #resize to mobile_net tensor size image = image - (103.94, 116.78, 123.68) image = image * (0.017, 0.017, 0.017) #change numpy data type as np.float32 to match tensor's format image = image.astype(np.float32) #Make var to save numpy input_var = MNN.expr.placeholder([1, 224, 224, 3], MNN.expr.NHWC) input_var.write(image) #cv2 read shape is NHWC, Module's need is NC4HW4, convert it input_var = MNN.expr.convert(input_var, MNN.expr.NC4HW4) #inference output_var = net.forward(input_var) #the output from net may be NC4HW4, turn to linear layout output_var = MNN.expr.convert(output_var, MNN.expr.NHWC) print("output belong to class: {}".format(np.argmax(output_var.read())))

你还，我在转换模型，包括onnx inference的时候没有任何问题：
`Start to Convert Other Model Format To MNN Model...
[10:59:27] /meida/shared/sda/fenghan/envs/MNN-master/tools/converter/source/onnx/onnxConverter.cpp:30: ONNX Model ir version: 6
Start to Optimize the MNN Net...
inputTensors : [ input, ]
outputTensors: [ scores, ]
Converted Success!`
但是再mnn inference的时候出现：
`Error for binary op: input0's type != input1's type
Compute Shape Error for scores
Can't run session because not resized
Segmentation fault`
通过逐步排查，发现问题出现在：
`class SimpleNMS(nn.Module):
    def __init__(self,nms_radius: int):
        super().__init__()
        self.maxpool_2d = nn.MaxPool2d(kernel_size=int(2*nms_radius + 1),stride = 1,padding=nms_radius)
    
    def forward(self,scores):
        scores = scores.unsqueeze(0)
        max_mask = self.maxpool_2d(scores) > 0.
        supp_mask = self.maxpool_2d(max_mask.float()) > 0
        # mask4 = max_mask > 0.
        return supp_mask`
改成：
`class SimpleNMS(nn.Module):
    def __init__(self,nms_radius: int):
        super().__init__()
        self.maxpool_2d = nn.MaxPool2d(kernel_size=int(2*nms_radius + 1),stride = 1,padding=nms_radius)
    
    def forward(self,scores):
        scores = scores.unsqueeze(0)
        max_mask = self.maxpool_2d(scores) > 0.
        return max_mask`
则一切正常，我不知道为什么会出现这种情况。
